src/ocm/components/clusterConfiguration/networkConfiguration/AvailableSubnetsControl.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect } from 'react';
+import {
+  Alert,
+  AlertVariant,
+  Button,
+  FormGroup,
+  Stack,
+  StackItem,
+  Tooltip,
+} from '@patternfly/react-core';
+import { FieldArray, useFormikContext } from 'formik';
+import { MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import { Cluster, Host } from '../../../../common/api/types';
+import { HostSubnets, NetworkConfigurationValues } from '../../../../common/types';
+import { NO_SUBNET_SET } from '../../../../common/config/constants';
+import { SelectField } from '../../../../common/components/ui';
+
+const REMOVE_BUTTON_EXIT_DELAY = 1500;
+
+export interface AvailableSubnetsControlProps {
+  clusterId: Cluster['id'];
+  hostSubnets: HostSubnets;
+  hosts: Host[];
+  isRequired: boolean;
+  isMultiNodeCluster: boolean;
+}
+
+export const AvailableSubnetsControl = ({
+  clusterId,
+  hostSubnets,
+  isRequired = false,
+}: AvailableSubnetsControlProps) => {
+  const { values, errors, setFieldValue } = useFormikContext<NetworkConfigurationValues>();
+
+  useEffect(() => {
+    if (values.machineNetworks && values.machineNetworks?.length < 1) {
+      setFieldValue(
+        'machineNetworks',
+        [
+          {
+            cidr: hostSubnets.length >= 1 ? hostSubnets[0].subnet : NO_SUBNET_SET,
+            clusterId: clusterId,
+          },
+        ],
+        true,
+      );
+    }
+  }, [clusterId, setFieldValue, hostSubnets, values.machineNetworks]);
+
+  return (
+    <FormGroup
+      label="Machine network"
+      labelInfo={values.stackType === 'dualStack' && 'Primary'}
+      fieldId="machine-networks"
+      isRequired
+    >
+      <FieldArray name="machineNetworks">
+        {({ push, remove }) => (
+          <Stack>
+            {values.machineNetworks?.map((machineNetwork, index) => {
+              return (
+                <StackItem key={index}>
+                  <Tooltip
```

Comment:
The remove button tooltip is duplicated three times - perhaps create a component WithRemoveButton that gets the child component of the Tooltip and returns a new component with the remove button tooltip?
Also - for more readability (it took me time to understand that the tooltip is where the remove button is...) you can implement the remove button like this:
 const [showRemoveButton, setShowRemoveButton] = React.useState(false);
  const updateShowRemoveButton = (value: boolean) => {
    if (!enableRemoveHost) {
      setShowRemoveButton(false);
    } else {
      setShowRemoveButton(value);
    }
  };
  return (
    <ComponentWithRemoveButton
      onMouseEnter={() => updateShowRemoveButton(true)}
      onMouseLeave={() => updateShowRemoveButton(false)}
    >
    This is how I implemented it in static ip.
    If we have the WithRemoveButton component I'll be able to reuse it

Reply:
I'm adding a new component for this but I'm not sure I understand the code you wrote here. Are you suggesting that I should replace the `Tooltip` with it? That would certainly work just as well when it comes to showing and hiding the button. But the tooltip also takes care of that as well as the positioning (which would be much more difficult to do on our own).

I'm hoping that by creating a separate component for this we can get away with using the _Tooltip black magic_ without confusing too many people. @batzionb

Reply:
I'm adding a new component for this but I'm not sure I understand the code you wrote here. Are you suggesting that I should replace the `Tooltip` with it? That would certainly work just as well when it comes to showing and hiding the button. But the tooltip also takes care of that as well as the positioning (which would be much more difficult to do on our own).

I'm hoping that by creating a separate component for this we can get away with using the _Tooltip black magic_ without confusing too many people. @batzionb

Reply:
sure, you can use the tooltip for it